src/OrchardCore.Modules/OrchardCore.Media.AmazonS3/CreateMediaS3BucketEvent.cs

```diff
@@ -14,74 +16,117 @@ namespace OrchardCore.Media.AmazonS3;
+            _shellSettings.State == Environment.Shell.Models.TenantState.Uninitialized ||
+            String.IsNullOrEmpty(_options.BucketName))
         {
-            _logger.LogDebug("Testing Amazon S3 Bucket {BucketName} existence", _options.BucketName);
+            return;
+        }
+
+        _logger.LogDebug("Testing Amazon S3 Bucket {BucketName} existence", _options.BucketName);
 
-            try
+        try
+        {
+            var bucketExists = await AmazonS3Util.DoesS3BucketExistV2Async(_amazonS3Client, _options.BucketName);
+            if (bucketExists)
             {
-                var isBucketExists = await AmazonS3Util.DoesS3BucketExistV2Async(_amazonS3Client, _options.BucketName);
-                if (isBucketExists)
-                {
-                    _logger.LogInformation("Amazon S3 Bucket {BucketName} already exists.", _options.BucketName);
-                    return;
-                }
-
-                var bucketRequest = new PutBucketRequest
-                {
-                    BucketName = _options.BucketName,
-                    UseClientRegion = true
-                };
-
-                // Tying to create bucket
-                var response = await _amazonS3Client.PutBucketAsync(bucketRequest);
-
-                if (!response.IsSuccessful())
-                {
-                    _logger.LogError("Unable to create Amazon S3 Bucket. {Response}", response);
-                    return;
-                }
-
-                // Blocking public access for the newly created bucket.
-                var blockConfiguration = new PublicAccessBlockConfiguration
-                {
-                    BlockPublicAcls = true,
-                    BlockPublicPolicy = true,
-                    IgnorePublicAcls = true,
-                    RestrictPublicBuckets = true
-                };
-
-                await _amazonS3Client.PutPublicAccessBlockAsync(new PutPublicAccessBlockRequest
-                {
-                    PublicAccessBlockConfiguration = blockConfiguration,
-                    BucketName = _options.BucketName
-                });
-
-                _logger.LogDebug("Amazon S3 Bucket {BucketName} created.", _options.BucketName);
+                _logger.LogInformation("Amazon S3 Bucket {BucketName} already exists.", _options.BucketName);
+                return;
             }
-            catch (Exception e)
+
+            var bucketRequest = new PutBucketRequest
             {
-                _logger.LogError(e, "Unable to create Amazon S3 Bucket.");
+                BucketName = _options.BucketName,
+                UseClientRegion = true
+            };
+
+            // Trying to create bucket.
+            var response = await _amazonS3Client.PutBucketAsync(bucketRequest);
+
+            if (!response.IsSuccessful())
+            {
+                _logger.LogError("Unable to create Amazon S3 Bucket {BucketName}", _options.BucketName);
+                return;
```

Comment:
```suggestion

                return;
```

Reply:
Okay, done on my side.

Normally not related to this PR, so to not add too much github diff. But anyway we see this part in the diff just because we change the pattern of the if block as done in the RemovingAsync() we added here.

---

Yes I also prefer to not have compact code but when code is spaced too evenly sometimes I prefer to remove some lines to see more easily different parts of the code, for example before I was not a fan to not add a line between a variable declaration and e.g. an if block, but now I like it as long as analyzers don't complain, for example I would have used.


            var response = await _amazonS3Client.PutBucketAsync(bucketRequest);
            if (!response.IsSuccessful())
            {
                _logger.LogError("Unable to create Amazon S3 Bucket {BucketName}", _options.BucketName);
                return;
            }


What I always do is to add a line after the last `}` of any block. Also add a line before a return if at the end or if before more than let's say 2 or 3 lines of code.